packages/superset-ui-chart/src/clients/ChartClient.ts

```diff
@@ -84,19 +84,34 @@ export default class ChartClient {
       const { useLegacyApi } = metaDataRegistry.get(visType)!;
       const buildQuery = (await buildQueryRegistry.get(visType)) ?? (() => formData);
 
-      return this.client
-        .post({
-          headers: { 'Content-Type': 'application/json' },
-          endpoint: useLegacyApi ? '/superset/explore_json/' : '/api/v1/chart/data',
-          postPayload: {
-            [useLegacyApi ? 'form_data' : 'query_context']: buildQuery(formData),
-          },
-          ...options,
-        } as RequestConfig)
-        .then(response => {
-          // let's assume response.json always has the shape of QueryData
-          return response.json as QueryData;
-        });
+      if (useLegacyApi) {
+        return this.client
+          .post({
+            ...options,
+            endpoint: '/superset/explore_json/',
+            postPayload: {
+              form_data: buildQuery(formData),
+            },
+          } as RequestConfig)
+          .then(({ json }) => json as QueryData);
+      }
+
+      return (
+        this.client
+          .post({
+            ...options,
+            endpoint: '/api/v1/chart/data',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify(buildQuery(formData)),
```

Comment:
I wonder if we can add a method `postJson` or option `post({ json: ... })` to `SupersetClient`. I'd imagine we are going to see more and more POST json requests.

Reply:
I think it'd be best to have a `SupersetClientV1` that uses conventions for the v1 api by default across the board.

Reply:
This will be refactored to use functions from #555.

Reply:
@suddjian If we want something like `SupersetClientV1`, then we should probably remove the flexibility of specifying arbitrary API endpoint.

@kristw I think a `postJson` or `post({ json: ... })` is still needed, for new API endpoints like seen here: https://github.com/apache/incubator-superset/pull/10104/files#r442641384 .

There should be two layer of abstraction: 1) a utility client that make requests to arbitrary endpoints 2) a true Superset client with schema enforced.

For now with all the unstable new APIs, people should start using 1, so they don't have to go back and forth editing two repos. After backend API becomes more stable, we migration them to 2.

@villebro @rusackas @etr2460  thoughts?